pkg/kv/dynamodb/store.go

```diff
@@ -0,0 +1,320 @@
+package dynamodb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/treeverse/lakefs/pkg/kv"
+)
+
+type Driver struct{}
+
+type Store struct {
+	svc    *dynamodb.DynamoDB
+	params *Params
+}
+
+type EntriesIterator struct {
+	entry        *kv.Entry
+	err          error
+	store        *Store
+	queryResult  *dynamodb.QueryOutput
+	currEntryIdx int
+	partKey      string
+	startKey     string
+}
+
+type DynKVItem struct {
+	PartKey   string
+	ItemKey   string
+	ItemValue string
+}
+
+const (
+	DriverName       = "dynamodb"
+	DefaultTableName = "kvstore"
+
+	PartKey   = "PartKey"
+	ItemKey   = "ItemKey"
+	ItemValue = "ItemValue"
+
+	// TBD: Which values to use?
+	ReadCapacityUnits  = 1000
+	WriteCapacityUnits = 1000
+)
+
+//nolint:gochecknoinits
+func init() {
+	kv.Register(DriverName, &Driver{})
+}
+
+func (d *Driver) Open(ctx context.Context, name string) (kv.Store, error) {
+	sess := session.Must(session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	}))
```

Comment:
Let's reuse the session we create for the blockstore. Unless you expect the credentials to be passed twice, they should be handled in a single place.

Reply:
Relevant credentials from env are read and overriden by aws conf files (if any).
In addition, added another layer to allow specific test params (fake access and secret keys) without affecting the env